tdmpc2/common/world_model.py

```diff
@@ -160,13 +160,14 @@ def Q(self, z, a, task, return_type='min', target=False):
 
 		if self.cfg.multitask:
 			z = self.task_emb(z, task)
-			
+
 		z = torch.cat([z, a], dim=-1)
 		out = (self._target_Qs if target else self._Qs)(z)
 
 		if return_type == 'all':
 			return out
 
-		Q1, Q2 = out[np.random.choice(self.cfg.num_q, 2, replace=False)]
+		rand_idx = torch.randperm(self.cfg.num_q)[:2].unbind(0)
+		Q1, Q2 = out[rand_idx[0]], out[rand_idx[1]]
```

Comment:
just a PT version of the above. Cudagraphs didn't like the auto-unbind like `Q1, Q2 = out[idx]` so I'm doing that manually